openedx/adg/lms/webinars/admin.py

```diff
@@ -71,52 +74,64 @@ def get_fields(self, request, obj=None):
         """
         fields = super().get_fields(request, obj)
         if not obj:
-            fields.remove('send_update_emails_to_registrants')
+            fields.remove('send_update_emails')
         return fields
 
     def save_related(self, request, form, formsets, change):
         """
         Extension of save_related for webinar to send emails when object is created or modified.
         """
+        new_members = []
+        removed_members = []
         if change and any(field in form.changed_data for field in ['co_hosts', 'presenter', 'panelists']):
-            update_webinar_team_registrations(form)
+            new_members, removed_members = get_newly_added_and_removed_team_members(form)
 
-        super(WebinarAdmin, self).save_related(request, form, formsets, change)
+        super().save_related(request, form, formsets, change)
 
         webinar = form.instance
 
-        webinar_invitation_recipients = form.cleaned_data.get('invites_by_email_address', [])
-        if form.cleaned_data.get('invite_all_platform_users'):
-            webinar_invitation_recipients += list(
-                User.objects.exclude(email='').values_list('email', flat=True)
-            )
+        webinar_invitees_emails = get_webinar_invitees_emails(form)
 
         if change:
-            registered_users = list(
-                webinar.registrations.filter(is_registered=True).values_list('user__email', flat=True)
-            )
-            if form.cleaned_data.get('send_update_emails_to_registrants'):
+            if removed_members:
+                remove_team_registrations_and_cancel_reminders(removed_members, webinar)
+
+            webinar_update_recipients_emails = []
+            if new_members or webinar_invitees_emails or form.cleaned_data.get('send_update_emails'):
+                webinar_update_recipients_emails = get_webinar_update_recipients_emails(webinar)
```

Comment:
Is there a chance that `webinar_update_recipients_emails` ends up as an empty list? (if there are no registrations eg)
In that case perhaps we should check before calling `send_webinar_emails`?

Reply:
Apparently no because presenter is a compulsory field in webinar. As long as there is a presenter, webinar_update_recipients will always have a a single value at least.